src/main/java/com/wix/mysql/EmbeddedMysql.java

```diff
@@ -30,7 +32,13 @@ protected EmbeddedMysql(final MysqldConfig config) {
         logger.info("Preparing EmbeddedMysql version '{}'...", config.getVersion());
         this.config = config;
         IRuntimeConfig runtimeConfig = new RuntimeConfigBuilder().defaults(config.getVersion()).build();
-        this.executable = new MysqldStarter(runtimeConfig).prepare(config);
+
+        lock.lock();
+        try {
+            this.executable = new MysqldStarter(runtimeConfig).prepare(config);
```

Comment:
Do you need to lock on both calls? I would try to extract locking in a helper method so it would interfere with MySql startup.